client/proposal.go

```diff
@@ -304,8 +304,7 @@ func (c *Client) validTwoPartyProposal(
 	ourIdx int,
 	peerAddr wallet.Address,
 ) error {
-	base := proposal.Base()
-	if err := base.Valid(); err != nil {
+	if err := proposal.Valid(); err != nil {
```

Comment:
check that all implementations check `base.Valid()`.

check that `Proposal.Valid` interface documentation requires calling of `base.Valid`.